### PR TITLE
Fix immediate renderer image and text rotation in rotated canvas

### PR DESCRIPTION
**Pull request by mike-000**
Fixes the inconsistent rotations observed in https://github.com/openlayers/openlayers/issues/14308#issuecomment-1321258691

Subtract canvas rotation from image or text rotation so it does not automatically rotate with the view (or double that rotation if `rotateWithView` is specified).

Replace circle style with triangle in rendering tests so any incorrect image rotation can be seen.

### CODE REVIEW

File: src/ol/render/canvas/Immediate.js [hunk 1]
```diff
@@ -83,6 +84,14 @@ class CanvasImmediateRenderer extends VectorContext {
      */
     this.transform_ = transform;
 
+    /**
+     * @private
+     * @type {number}
+     */
+    this.transformRotation_ = transform
```
Comment (ahocevar) [inline src/ol/render/canvas/Immediate.js:91]: Something looks wrong here. `transform` is a transform matrix, not a number.

Comment (mike-000) [reply src/ol/render/canvas/Immediate.js:91]: The rotation is the number.  I first check that the transform is not undefined or null (which is should never be) as several browser tests which create a CanvasImmediateRenderer without using `drawImages_` or `drawText_` fail if they attempt to access entries from a missing matrix.

File: src/ol/render/canvas/Immediate.js [hunk 2]
```diff
@@ -83,6 +84,14 @@ class CanvasImmediateRenderer extends VectorContext {
      */
     this.transform_ = transform;
```
Comment (ahocevar) [inline src/ol/render/canvas/Immediate.js:85]: Here you apply the transform...

File: src/ol/render/canvas/Immediate.js (hunk 1 above)
Comment (ahocevar) [inline src/ol/render/canvas/Immediate.js:91]: ... and here you apply the same transform, but the type says `number`. If you want the rotation here, you need to extract it from the transform, or assign null initially and note that in the type.

Comment (mike-000) [reply src/ol/render/canvas/Immediate.js:91]: That is not the end of the statement.  I check if the transform exists before extracting from it.

Comment (ahocevar) [reply src/ol/render/canvas/Immediate.js:91]: Oh I see. Sorry, my bad.

**Review by ahocevar — APPROVED** [review]
Thanks, @mike-000

**Pull request merged**